### PR TITLE
SVCPLAN-4965: add check_keytab postscript

### PR DESCRIPTION
**Pull request by jakerundall**

### CODE REVIEW

**Comment by jakerundall** [comment]
The check_keytab postscript makes sure that a keytab doesn't contain stale host principals. The goal is to identify this problem promptly on boot rather than finding out due to user reports, etc.

I've been testing this on MG since May.

File: postscripts/check_keytab [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+  fi
+
+  exit 99
+}
+
+
+print_usage() {
+    cat <<ENDHERE
+Usage:
+    puppet_run [OPTIONS] [ENV_NAME]
```
Comment (billglick) [inline postscripts/check_keytab]:
```suggestion
    check_keytab [OPTIONS]
```

File: postscripts/check_keytab [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+#
+PRG=$( basename $0 )
+logger -t xcat -p local4.info "running '$PRG' on node $NODE"
+
+
+#
+# DEFAULT SETTINGS
+#
+DEBUG=1
+PRINCIPAL="host/"`hostname -f`"@NCSA.EDU"
```
Comment (billglick) [inline postscripts/check_keytab:39]: I am worried that there may be cases where `hostname -f` intentionally doesn't match the hostname of the host's kerberos principal. I can't come up with an example at the moment, but I guess that is the point of optionally being able to pass in the host principal name.

No change needed. Just thinking out loud.

**Review by billglick — CHANGES_REQUESTED** [review]
The name of the postscript is wrong in the --help option.

**Comment by billglick** [comment]
There was a part of me that questioned if we should move bad keytab file aside so that theoretically Puppet could fix it. But after thinking about that for a few minutes, I realized some manual clean up will almost always be required if that is the case.

File: postscripts/check_keytab (hunk 1 above)
Comment (jakerundall) [reply postscripts/check_keytab]: Fixed, thanks!

**Comment by jakerundall** [comment]
> There was a part of me that questioned if we should move bad keytab file aside so that theoretically Puppet could fix it. But after thinking about that for a few minutes, I realized some manual clean up will almost always be required if that is the case.

It definitely _could_ do that. And we could also update Puppet to delete any host principal that already exists in that case so it can recreate it. However I think we generally use createhost principals, which can only create and not delete AFAIK, for Puppet. But something to think about.

**Review by billglick — APPROVED** [review]

**Pull request merged**